contracts/DomainRegistar.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+// Uncomment this line to use console.log
+// import "hardhat/console.sol";
+
+error AccessDenied(string description);
+error NotEnoughFunds(uint8 provided, uint8 required);
+error DuplicateDomain();
+error TopLevelDomainsOnly();
+
+/**
+ * @title A simple contract for top-level domain registration
+ * @author Me
+ */
+contract DomainRegistar {
```

Comment:
// contract in general 
// no logic to withdraw founds from contract
// - 20 points